core/libraries/rest_api/RestIncomingQueryParamMetadata.php

```diff
@@ -0,0 +1,705 @@
+        ));
+        // double-check is it a *_gmt field?
+        if (!$this->getField() instanceof EE_Model_Field_Base
+            && ModelDataTranslator::isGmtDateFieldName($this->getQueryParamKeySansStars())
+        ) {
+            // yep, take off '_gmt', and find the field
+            $this->setQueryParamKey(ModelDataTranslator::removeGmtFromFieldName($this->getQueryParamKeySansStars()));
+            $this->setField(ModelDataTranslator::deduceFieldFromQueryParam(
+                $this->getQueryParamKey(),
+                $this->context->getModel()
+            ));
+            $this->setTimezone('UTC');
+            $this->setIsGmtField(true);
+        } elseif ($this->getField() instanceof EE_Datetime_Field) {
+            // so it's not a GMT field. Set the timezone on the model to the default
+            $this->setTimezone(EEH_DTT_Helper::get_valid_timezone_string());
+        } else {
+            // just keep using what's already set for the timezone
+            $this->setTimezone($this->context->getModel()->get_timezone());
+        }
+    }
+
+    /**
+     * Given a ton of input, determines the value to use for the models.
+     * @since $VID:$
+     * @return array|null
+     * @throws DomainException
+     * @throws EE_Error
+     * @throws RestException
+     * @throws DomainException
+     */
+    public function determineConditionsQueryParameterValue()
+    {
+        if ($this->valueIsArrayDuringRead()) {
+            $translated_value = $this->determineModelValueGivenRestInputArray();
+        } else {
+            $translated_value = ModelDataTranslator::prepareFieldValueFromJson(
+                $this->getField(),
+                $this->getQueryParamValue(),
+                $this->getContext()->getRequestedVersion(),
+                $this->getTimezone()
+            );
+        }
+        return $translated_value;
+    }
+
+    /**
+     * Given that the array value provided was itself an array, handles finding the correct value to pass to the model.
+     * @since $VID:$
+     * @return array|null
+     * @throws RestException
+     */
+    private function determineModelValueGivenRestInputArray()
+    {
+        $this->transformSimplifiedSpecifiedOperatorSyntaxIntoStandardSyntax();
+        // did they specify an operator?
+        if ($this->valueIsLegacySpecifiedOperator()) {
+            $query_param_value = $this->getQueryParamValue();
+            $sub_array_key = $query_param_value[0];
+            $translated_value = array($sub_array_key);
+            if ($this->operatorIsNAry($sub_array_key)) {
+                $translated_value[] = $this->prepareValuesFromJson($query_param_value[1]);
+            } elseif ($this->operatorIsTernary($sub_array_key)) {
+                $translated_value[] = array(
+                    $this->prepareValuesFromJson($query_param_value[1][0]),
+                    $this->prepareValuesFromJson($query_param_value[1][1])
+                );
```

Comment:
this is soooooooooooo much easier to read now, good job!

Reply:
thanks!